### DISCUSSION TITLE
Is there a way to parse the contents of the txt file uploaded to the web server?

### DISCUSSION THREAD

**Discussion by xczsonn** [rejetto/hfs#484] [Q&A]
Is there a way to parse the contents of the txt file uploaded to the web server? I used BeautifulSoup when I used other web server programs, but it doesn't seem to work with this program. It's a code that I wrote.

`
    @p.subcommand(description="가장 최신의 로그 파일 내용 보기")
    async def log(self, interaction: nextcord.Interaction):
      await interaction.response.defer(ephemeral=True)
      async with httpx.AsyncClient() as client:
        try:
            r = await client.get(self.log_url)
            soup = BeautifulSoup(r.content, 'html.parser')

            links = soup.find_all('a', href=True)
            latest_file = None
            latest_file_time = datetime.min.replace(tzinfo=None)

            for link in links:
                href = link['href']
                if 'txt' in href:
                    file_name = urllib.parse.unquote(href)
                    file_date_str = file_name.replace('.txt', '').replace('%2d', '-').replace('-', ' ')

                    try:
                        file_time = datetime.strptime(file_date_str, "%d %m %Y %H %M %S")
                        file_time = file_time.replace(tzinfo=None) + timedelta(hours=9)
                        if file_time > latest_file_time:
                            latest_file_time = file_time
                            latest_file = href
                    except ValueError as e:
                        print(f"Error parsing date for {file_name}: {e}")
                        continue

            if latest_file:
                file_url = self.log_url + latest_file
                file_r = await client.get(file_url)
                content = file_r.text

                # 추가된 필터링 문장들
                exclude_phrases = [
                    "Unhandled RequestConcreteModel_Server_FPalNetArchive",
                    "Registered chat",
                    "if you have paid for this",
                    "Discord",
                    "discord",
                    "PalGuard is starting",
                    "isPVPEnabled",
                    "Added neutral guid"
                ]
                
                content_lines = content.split('\n')
                filtered_content_lines = [line for line in content_lines if not any(phrase in line for phrase in exclude_phrases)]
                filtered_content = '\n'.join(filtered_content_lines)

                if len(filtered_content) > 1900:
                    for i in range(0, len(filtered_content), 1900):
                        part_content = filtered_content[i:i+1900]
                        if interaction.response.is_done():
                            await interaction.followup.send(f"```{part_content}```", ephemeral=True)
                        else:
                            await interaction.response.send_message(f"```{part_content}```", ephemeral=True)
                else:
                    if interaction.response.is_done():
                        await interaction.followup.send(f"```{filtered_content}```", ephemeral=True)
                    else:
                        await interaction.response.send_message(f"```{filtered_content}```", ephemeral=True)
            else:
                if interaction.response.is_done():
                    await interaction.followup.send("로그 파일을 찾을 수 없습니다.", ephemeral=True)
                else:
                    await interaction.response.send_message("로그 파일을 찾을 수 없습니다.", ephemeral=True)
        except Exception as e:
            error_message = f"로그 파일을 검색하는 도중 오류가 발생했습니다: {e}"
            print(error_message)
            if interaction.response.is_done():
                await interaction.followup.send(error_message, ephemeral=True)
            else:
                await interaction.response.send_message(error_message, ephemeral=True)
`

**Comment by rejetto** [comment]
i'm sorry, i forgot about this message.
I see you are using python. HFS offers no support for python, but you can consider using javascript to run python.
Javascript on the server-side is executed by nodejs, so that's your API.
